src/lib/journey/journey.interfaces.ts

```diff
@@ -64,6 +64,7 @@ export interface StepMetadata {
     numOfCallbacks: number;
     numOfSelfSubmittableCbs: number;
     numOfUserInputCbs: number;
+    stageName?: string;
```

Comment:
I'm hoping I can remove this.

Reply:
I'm guilty of this as well, but I think theres a few cases now where we are adding optional properties to interfaces and the interfaces over time are becoming less strict as a result. I think we may be better off auditing this problem in the future and creating discriminated unions for the various types. This will help allow narrowing and better saftey overall.

nothing for this pr, but a thought.